hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ExitCallsEnricher.java

```diff
@@ -61,19 +61,37 @@ public void enrichTrace(StructuredTrace trace) {
     // event -> api exit call count for the corresponding api_node
     Map<ByteBuffer, ApiExitCallInfo> eventToExitInfo = Maps.newHashMap();
 
+    int totalTraceExitCallCount = 0;
+
     for (ApiNode<Event> apiNode : apiTraceGraph.getApiNodeList()) {
       List<Event> backendExitEvents =
           apiTraceGraph.getExitBoundaryEventsWithNoOutboundEdgeForApiNode(apiNode);
       List<ApiNodeEventEdge> edges = apiTraceGraph.getOutboundEdgesForApiNode(apiNode);
       int totalExitCallCount = backendExitEvents.size() + edges.size();
       ApiExitCallInfo apiExitCallInfo = new ApiExitCallInfo().withExitCallCount(totalExitCallCount);
+      totalTraceExitCallCount += totalExitCallCount;
       edges.forEach(edge -> apiExitCallInfo.handleApiNodeEdge(trace, edge));
       backendExitEvents.forEach(apiExitCallInfo::handleBackend);
       apiNode.getEvents().forEach(e -> eventToExitInfo.put(e.getEventId(), apiExitCallInfo));
     }
+
+    addTotalNumberOfApiCallsEnrichedAttribute(apiTraceGraph, totalTraceExitCallCount);
+
     return eventToExitInfo;
   }
 
+  private void addTotalNumberOfApiCallsEnrichedAttribute(
+      ApiTraceGraph apiTraceGraph, int totalTraceExitCallCount) {
+    if (!apiTraceGraph.getApiNodeList().isEmpty()
+        && apiTraceGraph.getApiNodeList().get(0).getHeadEvent() != null
+        && totalTraceExitCallCount > 0) {
+      addEnrichedAttribute(
+          apiTraceGraph.getApiNodeList().get(0).getHeadEvent(),
```

Comment:
Shouldn't we enrich the trace, instead of the head span?

Reply:
Per original request, this should be a head span

Reply:
Agree with @skjindal93 
Logically it makes sense to have this as a trace attribute. If any downstream service needs it as a part of the head span its their responsibility to copy it wherever required.